src/utils/utilsTests/searchUtils.test.ts

```diff
@@ -51,6 +51,53 @@ describe("searchUtils", () => {
         })
       ).toBe(true)
     })
+    it("includes advanced search query params when field is set to 'all'", () => {
+      const testQuery =
+        "?q=shel%20silverstein&contributor=shel silverstein&title=the giving tree&subject=books"
+      expect(
+        checkQueryParamsEquality(testQuery, {
+          q: "shel silverstein",
+          title: "the giving tree",
+          contributor: "shel silverstein",
+          subject: "books",
+          field: "all",
+        })
+      ).toBe(true)
+    })
+    it("clears advanced search query params when field param is anything other than 'all'", () => {
+      const titleQuery = "?q=shel%20silverstein&search_scope=title"
+      expect(
+        checkQueryParamsEquality(titleQuery, {
+          q: "shel silverstein",
+          title: "the giving tree",
+          contributor: "shel silverstein",
+          subject: "books",
```

Comment:
I'm confused. I thought that when there's a field other than "all", then title, contributor, and subject would be clear and not in the returned query string?

Reply:
Ah, never mind. I was not reading the helper test function correctly.